src/reference-manual/pedantic-mode.Rmd

```diff
@@ -0,0 +1,166 @@
+In general, assigning a parameter a uniform distribution can create non-differentiable boundary conditions and is not recommended.
+
+
+#### (Inverse-) Gamma distributions
+
+Gamma distributions are sometimes used as an attempt to assign an improper prior to a parameter.
+Pedantic mode gives a warning when the Gamma arguments indicate that this may be the case.
+
+
+#### lkj\_corr distribution
+
+Any use of the `lkj_corr` distribution generates a warning that suggests using the Cholesky variant instead.
+See <https://mc-stan.org/docs/functions-reference/lkj-correlation.html> for details.
+
+
+## Unused parameters
+
+A warning is generated when a parameter is declared but does not have any effect on the program.
+This is determined by checking whether the value of the `target` variable depends in any way on each of the parameters.
+
+
+## Large or small constants in a distribution
+
+When numbers with magnitude less than 0.1 or greater than 10 are used as arguments to a distribution, it indicates that some parameter is not scaled to unit value, so a warning is thrown.
+See <https://mc-stan.org/docs/stan-users-guide/standardizing-predictors-and-outputs.html> for a discussion of scaling parameters.
+
+
+## Control flow depends on a parameter
+
+Control flow statements, such as `if`, `for` and `while`, should not depend on the value of the parameters to determine their branching conditions.
+Otherwise, the program may branch differently between iterations, which is likely to introduce discontinuity into the density function.
+Pedantic mode generates a warning when any branching condition may depend on a parameter value.
+
+
+## Parameters with multiple twiddles
+
+A warning is generated when a parameter is found on the left-hand side of more than one ~ statements (or an equivalent `target +=` conditional density statement).
+This pattern is not inherently an issue, but it is unusual and may indicate a mistake.
+
+Pedantic mode only searches for repeated statements, it will not for example generate a warning when a ~ statement is executed repeatedly inside of a loop.
+
+
+## Parameters with zero or multiple priors
+
+A warning is generated when a parameter appears to have greater than or less than one prior distribution factor.
+
+This analysis depends on a [*factor graph*](https://en.wikipedia.org/wiki/Factor_graph) representation of a Stan program. A factor F that depends on a parameter P is called a *prior factor for P* if there is no path in the factor graph from F to any data variable except through P.
+
+
+## Variables used before assignment
+
+A warning is generated when any variable is used before it has been assigned a value.
+This warning is also available as a standalone option to Stanc3, with the flag: `--warn-uninitialized`.
+
+
+## Strict or nonsensical parameter bounds
+
+Parameters that are have strict `upper` and `lower` bounds can cause unmanageably large gradients in a density function, and may only be justified in a few cased.
```

Comment:
From the error message itself, I think "Except when there are logical or physical constraints, it is very unusual for you to be sure that a parameter will fall inside a specified range" is more important than the infinite gradients.